lib/registry.ex

```diff
@@ -318,8 +320,10 @@ defmodule Registry do
   associated to the pid. If there are no entries for the given key,
   the callback is never invoked.
 
-  If the registry is partitioned, the callback will be invoked for
-  every partition that has matching entries **concurrently**.
+  If the registry is not partitioned, the callback is invoked in the process
+  that calls `dispatch/3`. If the registry is partitioned, the callback is
+  invoked for every partition that has matching entries **concurrently** (and
+  will run in each partition).
```

Comment:
I think we should remove the parens or word it better. It feels a bit "lost". I am not quite sure it means. :)